File: frontend/src/api/recruitment.ts
import { apiClient } from './client'
import { JobPostingDto, CandidateDto, InterviewDto, JobOfferDto } from '@/types'

export const recruitmentApi = {
  // Job Postings
  getJobPostings: (status?: 'Open' | 'Closed') => {
    const url = status ? `/recruitment/job-postings?status=${status}` : '/recruitment/job-postings'
    return apiClient.get<JobPostingDto[]>(url)
  },

  getJobPostingById: (id: string) =>
    apiClient.get<JobPostingDto>(`/recruitment/job-postings/${id}`),

  createJobPosting: (data: {
    title: string
    description: string
    departmentId: string
  }) =>
    apiClient.post('/recruitment/job-postings', data),

  closeJobPosting: (id: string) =>
    apiClient.put(`/recruitment/job-postings/${id}/close`, {}),

  // Candidates
  getCandidates: (jobPostingId?: string) => {
    const url = jobPostingId 
      ? `/recruitment/candidates?jobPostingId=${jobPostingId}` 
      : '/recruitment/candidates'
    return apiClient.get<CandidateDto[]>(url)
  },

  getCandidateById: (id: string) =>
    apiClient.get<CandidateDto>(`/recruitment/candidates/${id}`),

  createCandidate: (data: {
    firstName: string
    lastName: string
    email: string
    phoneNumber?: string
    jobPostingId: string
  }) =>
    apiClient.post('/recruitment/candidates', data),

  updateCandidateStatus: (id: string, status: string) =>
    apiClient.put(`/recruitment/candidates/${id}/status`, { status }),

  // Interviews
  getInterviews: (candidateId?: string) => {
    const url = candidateId 
      ? `/recruitment/interviews?candidateId=${candidateId}` 
      : '/recruitment/interviews'
    return apiClient.get<InterviewDto[]>(url)
  },

  scheduleInterview: (data: {
    candidateId: string
    scheduledDateUtc: string
    interviewerName: string
  }) =>
    apiClient.post('/recruitment/interviews', data),

  completeInterview: (id: string, data: {
    rating: number
    feedback: string
  }) =>
    apiClient.put(`/recruitment/interviews/${id}/complete`, data),

  // Job Offers
  getJobOffers: (candidateId?: string) => {
    const url = candidateId 
      ? `/recruitment/offers?candidateId=${candidateId}` 
      : '/recruitment/offers'
    return apiClient.get<JobOfferDto[]>(url)
  },

  createJobOffer: (data: {
    candidateId: string
    offeredSalary: number
    expiryDateUtc: string
  }) =>
    apiClient.post('/recruitment/offers', data),

  acceptJobOffer: (id: string) =>
    apiClient.put(`/recruitment/offers/${id}/accept`, {}),

  rejectJobOffer: (id: string) =>
    apiClient.put(`/recruitment/offers/${id}/reject`, {}),
}

File: EmployeeLifecyclePortal.Domain/Entities/Recruitment.cs
using EmployeeLifecyclePortal.Domain.Common;

namespace EmployeeLifecyclePortal.Domain.Entities;

/// <summary>Sprint 24: Recruitment - Job postings, candidates, interviews, offers</summary>
public class JobPosting : AuditableEntity
{
    public string Title { get; private set; } = string.Empty;
    public string Description { get; private set; } = string.Empty;
    public Guid DepartmentId { get; private set; }
    public string Status { get; private set; } = "Open";
    public DateTime PostedDateUtc { get; private set; }
    public DateTime? ClosedDateUtc { get; private set; }
    public Department? Department { get; private set; }

    private JobPosting() { }

    public JobPosting(string title, string description, Guid departmentId)
    {
        Title = title;
        Description = description;
        DepartmentId = departmentId;
        PostedDateUtc = DateTime.UtcNow;
    }

    public void Close()
    {
        Status = "Closed";
        ClosedDateUtc = DateTime.UtcNow;
    }
}

public class Candidate : AuditableEntity
{
    public string FirstName { get; private set; } = string.Empty;
    public string LastName { get; private set; } = string.Empty;
    public string FullName => $"{FirstName} {LastName}".Trim();
    public string Email { get; private set; } = string.Empty;
    public string? PhoneNumber { get; private set; }
    public string Status { get; private set; } = "Applied";
    public Guid JobPostingId { get; private set; }
    public string? ResumePath { get; private set; }
    public JobPosting? JobPosting { get; private set; }

    private Candidate() { }

    public Candidate(string firstName, string lastName, string email, Guid jobPostingId)
    {
        FirstName = firstName;
        LastName = lastName;
        Email = email;
        JobPostingId = jobPostingId;
    }

    public void MoveToInterview()
    {
        Status = "Interview";
    }

    public void Reject()
    {
        Status = "Rejected";
    }

    public void UpdateStatus(string status)
    {
        Status = status;
    }

    public void SetPhoneNumber(string? phoneNumber)
    {
        PhoneNumber = phoneNumber;
    }
}

public class Interview : AuditableEntity
{
    public Guid CandidateId { get; private set; }
    public DateTime ScheduledDateUtc { get; private set; }
    public string InterviewerName { get; private set; } = string.Empty;
    public string Status { get; private set; } = "Scheduled";
    public int? Rating { get; private set; }
    public string? Feedback { get; private set; }
    public Candidate? Candidate { get; private set; }

    private Interview() { }

    public Interview(Guid candidateId, DateTime scheduledDate, string interviewerName)
    {
        CandidateId = candidateId;
        ScheduledDateUtc = scheduledDate;
        InterviewerName = interviewerName;
    }

    public void Complete(int rating, string feedback)
    {
        Status = "Completed";
        Rating = rating;
        Feedback = feedback;
    }
}

public class JobOffer : AuditableEntity
{
    public Guid CandidateId { get; private set; }
    public decimal OfferedSalary { get; private set; }
    public DateTime OfferDateUtc { get; private set; }
    public DateTime? ExpiryDateUtc { get; private set; }
    public string Status { get; private set; } = "Pending";
    public Candidate? Candidate { get; private set; }

    private JobOffer() { }

    public JobOffer(Guid candidateId, decimal salary, DateTime expiryDate)
    {
        CandidateId = candidateId;
        OfferedSalary = salary;
        OfferDateUtc = DateTime.UtcNow;
        ExpiryDateUtc = expiryDate;
    }

    public void Accept()
    {
        Status = "Accepted";
    }

    public void Reject()
    {
        Status = "Rejected";
    }
}
